app/javascript/components/lesson-preview/index.jsx

```diff
@@ -49,6 +48,12 @@ const LessonPreview = () => {
     }
   }, [copied]);
 
+  useEffect(() => {
```

Comment:
I'm a little nervous about doing this on every re-render, it's not a big deal at the moment but could become a performance problem as this feature grows.

This feels like it would be better suited to being executed through the onChange event in the text area so its only scoped to when that event fires.

We could add a `handleContentChange` function that could do what this useEffect is doing and `setContent` and then pass that through as the onChange prop for the `LessonContentInput` component instead.

Whats your thoughts?

Reply:
I can try that out, I figured that the component re-renders every time an onChange occurs, but now realize when the tab switches the component also re-renders. I initially had something similar to what you're suggesting, but it was not reliably updating the URL in state. It would update the `content` in state, but the URL in state would not always "catch up", so `useEffect` was my fix.

Reply:
ahh in that case we can pass an argument `[content]` argument to the useEffect so its only executed when the content state is updated https://reactjs.org/docs/hooks-effect.html#tip-optimizing-performance-by-skipping-effects

Reply:
Thanks for the link, I'll see what I can do :)